packages/contracts-bedrock/scripts/DeployDisputeGame.s.sol

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Forge
+import { Script } from "forge-std/Script.sol";
+
+// Scripts
+import { BaseDeployIO } from "scripts/utils/BaseDeployIO.sol";
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+
+// Libraries
+import { GameType, Claim, Duration } from "src/dispute/lib/Types.sol";
+import { LibString } from "@solady/utils/LibString.sol";
+
+// Interfaces
+import { IFaultDisputeGame } from "src/dispute/interfaces/IFaultDisputeGame.sol";
+import { IPermissionedDisputeGame } from "src/dispute/interfaces/IPermissionedDisputeGame.sol";
+import { IDelayedWETH } from "src/dispute/interfaces/IDelayedWETH.sol";
+import { IBigStepper, IPreimageOracle } from "src/dispute/interfaces/IBigStepper.sol";
+import { IAnchorStateRegistry } from "src/dispute/interfaces/IAnchorStateRegistry.sol";
+
+/// @title DeployDisputeGameInput
+contract DeployDisputeGameInput is BaseDeployIO {
+    // Specify which game kind is being deployed here.
+    string internal _gameKind;
+
+    // All inputs required to deploy FaultDisputeGame.
+    uint256 internal _gameType;
+    bytes32 internal _absolutePrestate;
+    uint256 internal _maxGameDepth;
+    uint256 internal _splitDepth;
+    uint256 internal _clockExtension;
+    uint256 internal _maxClockDuration;
+    IBigStepper internal _gameVm;
+    IDelayedWETH internal _delayedWethProxy;
+    IAnchorStateRegistry internal _anchorStateRegistryProxy;
+    uint256 internal _l2ChainId;
+
+    // Additional inputs required to deploy PermissionedDisputeGame.
+    address internal _proposer;
+    address internal _challenger;
+
+    function set(bytes4 _sel, uint256 _value) public {
+        if (_sel == this.gameType.selector) {
+            require(_value <= type(uint32).max, "DeployDisputeGame: gameType must fit inside uint32");
```

Comment:
I took this pattern of making everything a `uint256` from `DeployImplementations.s.sol` but it's unclear to me why this is needed. Can I use `uint32` instead? cc @mds1

Reply:
This was to reduce the number of `set` overloads. It also simplifies UX, because if you have `set(bytes4,uint256)` and `set(bytes4,uint32)` I think solidity complains if you call `set(myBytes4,myUint32)` because it can't tell which `set` method to use

Reply:
Gotcha, makes sense.